docs/overview/component/index-manager.md

```diff
@@ -0,0 +1,43 @@
+# Vald Index Manager
+
+Vald Index Manager is the component that controls the indexing process of Vald Agent Pods in the same Vald cluster.
+
+## Responsibility
+
+Vald Index Manager has a unique and simple role in preventing stop-the-world.
+
+It requires `Vald Discoverer` to fulfill its responsibility.
+
+## Features
+
+This chapter shows the main features to fulfill Vald Index Manager’s role.
+
+### Syncing data from Vald Discoverer
+
+Vald Index Manager gets the IP addresses of each Vald Agent pod from Vald Discoverer when its container starts.
+
+In addition, when IP address changes, Vald Index Manager gets the new IP address from Vald Discoverer.
+
+Vald Index Manager uses these for the controlling indexing process.
+
+### Controlling Indexing process
+
+When Vald Agent Pod creates or saves indexes on its container memory, it can not run the search process with the request query.
+
+Stop-the-world happens when all Vald Agent pods run the function involved in the indexing operation, e.g., `createIndex`, simultaneously.
+
+Vald Index Manager manages the indexing process of all Vald Agent pods to prevent this event.
+
+Vald Index Manager uses a Vald Agent pods' IP address list from Vald Discoverer and index information, including the stored index count, uncommitted index count, creating index count, and saving index count, from each Vald Agent pod.
+
+The control process is Vald Index Manager sends `createIndex` requests for concurrency simultaneously, sends a new request when the job is finished and continues until it sends to all agents.
+
+At the end of each process, Vald Index Manager updates the index information from each Vald Agent pod.
+
+Vald Index Manager runs this process repeatedly at set time intervals.
```

Comment:
```suggestion
Vald Index Manager runs this process periodically and when the number of uncommitted indexes exceeds the defined threshold.
```

Reply:
I suppose `when the number ~` is not correct :thinking: 
please see: https://github.com/vdaas/vald/blob/8377913a40ebc859fde5eb00e26a29fb8d19f9b7/pkg/manager/index/service/indexer.go#L197-L246

`execute` function does not check the uncommitted indexes with the threshold.
So, If I fix it, I'll update it like below.
```suggestion
Vald Index Manager runs this process periodically by set time intervals.
```

Reply:
Yes, there are no uncommitted index threshold, sorry for my mistake.

`periodically` means that it execute by some intervals, I think we can omit the `set time intervals` part.
```suggestion
Vald Index Manager runs this process periodically.
```

Reply:
well, we can.
It may be redundant, but we consider which is user-friendly because the user should control their time interval by setting their `valdrelease.yaml` even your suggestion is ok.
I just want your consideration.

Reply:
I think you are right.